Client/Application/AppDelegate.swift

```diff
@@ -158,3 +158,11 @@ extension AppDelegate: MFMailComposeViewControllerDelegate {
     }
 }
 #endif
+
+extension AppDelegate: UIApplicationDelegate {
+    func applicationDidEnterBackground(application: UIApplication) {
+        let notificationCenter = NSNotificationCenter.defaultCenter()
+        var info = [NSObject: AnyObject]()
+        notificationCenter.postNotificationName("ApplicationWillTerminate", object: self, userInfo: info)
```

Comment:
But please read my comment on the commit about using the system provided state preservation/restoration APIs. I don't think this is the right approach.